my-app/Dockerfile

```diff
@@ -0,0 +1,15 @@
+
+FROM node:current-alpine
+
+RUN apk add --update tini \
+    && mkdir -p /usr/src/app
+
+COPY package.json .
+
+COPY . .
```

Comment:
From what I've heard, installing the dependencies first, "compiling" the image and then attaching the rest of the source code is a good practice - the idea is that dependencies change less often than the code, so once installed, the layer with the dependencies installed will be more stable and re-used. I think the [multi-stage builds](https://docs.docker.com/develop/develop-images/multistage-build/) are related and used in that. I've seen such a niceky composed file generated for a container project in visual studio.